src/core/positioning/internalgnssreceiver.h

```diff
@@ -45,6 +45,8 @@ class InternalGnssReceiver : public AbstractGnssReceiver
     void handleConnectDevice() override;
     void handleDisconnectDevice() override;
 
+    bool mPermissionChecked = false;
```

Comment:
```suggestion
    bool mPermissionGranted = false;
```

Would that be more appropriate (i.e. checked and revoked will keep this set to "false")?